bdai_ros2_wrappers/bdai_ros2_wrappers/tf_listener_wrapper.py

```diff
@@ -119,36 +63,40 @@ def lookup_a_tform_b(
             frame_b: Tip frame for transform.  The transform returned will be frame_a_t_frame_b
             transform_time: The time at which to look up the transform.  If left at None, will be the most recent
                 transform available
-            timeout: The time to wait for the transform to become available if the transform_time is beyond the most
-                recent transform in the buffer.
+            timeout_sec: The time to wait for the transform to become available if the requested time is beyond
+                the most recent transform in the buffer. If set to 0, it will not wait. If left at None, it will
+                wait indefinitely.
             wait_for_frames: If true, waits timeout amount of time for a path to exist from frame_a to frame_b in the
-                buffer.  If false, this will return immediately if a path does not exist even if timeout is not
-                None.  Note that wait_for_transform can be used to wait indefinitely for a transform to become
-                available.
+                buffer. If false, this will return immediately if a path does not exist even if timeout is not
+                None. Note that wait_for_a_tform_b can also be used to wait for a transform to become available.
         Returns:
             The transform frame_a_t_frame_b at the time specified.
         Raises:
             All the possible TransformExceptions.
         """
-        return self._internal_lookup_a_tform_b(frame_a, frame_b, transform_time, timeout, wait_for_frames)
+        if not wait_for_frames:
+            timeout_sec = 0.0
```

Comment:
> if the path doesn't exist, it should return immediately if wait_for_frames is true.

Hmm, that is not what the original implementation does.

> I think the way this is written it will return immediately if wait_for_frames is true and transform_time is beyond what's in the tf buffer even if the path exists from frame a to frame b.

Hmm, it shouldn't. Didn't we test this a couple weeks ago? In fact, it's because it waits that the `test_future_transform_wait` test isn't flaky due to races.

> this is useful for code that may be adding frames later in the processing and therefore if the path doesn't exist, waiting isn't going to help.

I see. It's a bit of an odd use case, as this is a problem you would run into only if you had a single-threaded, synchronous program where statements are not ordered properly. That said, we can make it behave that way. Are you sure you want it to behave that way? It will expose us to races when using lookups not guarded by waits.

Reply:
Wow ok I did not write this correctly:

If `wait_for_frames` is _False_ then this should return immediately if a path from frame_a to frame_b does not exist.  Otherwise it should wait for the transform to exist until the timeout (or forever if the timeout is None), which I think is the normal thing the underlying TF stuff does.

We could also potentially drop this argument as I'm not really sure anyone is using it.